File: app/src/main/java/com/bari_ikutsu/coloredstatusbar/MainActivity.java
package com.bari_ikutsu.coloredstatusbar;

import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;

public class MainActivity extends ActionBarActivity {

    Button buttonChangeToRed, buttonChangeToGreen, buttonChangeToBlue;
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // ToolbarをActionBarに設定
        toolbar = (Toolbar) findViewById(R.id.tool_bar);
        setSupportActionBar(toolbar);

        // ボタンの取得
        buttonChangeToRed = (Button)findViewById(R.id.button);
        buttonChangeToGreen = (Button)findViewById(R.id.button2);
        buttonChangeToBlue = (Button)findViewById(R.id.button3);

        // ボタンのイベントハンドラ
        buttonChangeToRed.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                toolbar.setBackgroundColor(getResources().getColor(R.color.actionbar_red));
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                    getWindow().setStatusBarColor(getResources().getColor(R.color.statusbar_red));
                    getWindow().setNavigationBarColor(getResources().getColor(R.color.navigationbar_red));
                }
            }
        });

        buttonChangeToGreen.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                toolbar.setBackgroundColor(getResources().getColor(R.color.actionbar_green));
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                    getWindow().setStatusBarColor(getResources().getColor(R.color.statusbar_green));
                    getWindow().setNavigationBarColor(getResources().getColor(R.color.navigationbar_green));
                }
            }
        });

        buttonChangeToBlue.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                toolbar.setBackgroundColor(getResources().getColor(R.color.actionbar_blue));
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                    getWindow().setStatusBarColor(getResources().getColor(R.color.statusbar_blue));
                    getWindow().setNavigationBarColor(getResources().getColor(R.color.navigationbar_blue));
                }
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/bari_ikutsu/coloredstatusbar/MainActivity.kt
package com.bari_ikutsu.coloredstatusbar

import android.os.Build
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // ToolbarをActionBarに設定
        val toolbar = findViewById(R.id.tool_bar) as Toolbar
        setSupportActionBar(toolbar)

        // ボタンのイベントハンドラ
        button_red.setOnClickListener {
            toolbar.setBackgroundColor(ContextCompat.getColor(this, R.color.actionbar_red))
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                window.statusBarColor = ContextCompat.getColor(this, R.color.statusbar_red)
                window.navigationBarColor = ContextCompat.getColor(this, R.color.navigationbar_red)
            }
        }

        button_green.setOnClickListener {
            toolbar.setBackgroundColor(ContextCompat.getColor(this, R.color.actionbar_green))
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                window.statusBarColor = ContextCompat.getColor(this, R.color.statusbar_green)
                window.navigationBarColor = ContextCompat.getColor(this, R.color.navigationbar_green)
            }
        }

        button_blue.setOnClickListener {
            toolbar.setBackgroundColor(ContextCompat.getColor(this, R.color.actionbar_blue))
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                window.statusBarColor = ContextCompat.getColor(this, R.color.statusbar_blue)
                window.navigationBarColor = ContextCompat.getColor(this, R.color.navigationbar_blue)
            }
        }
    }
}
